documentation/guides/draft/use-cases/end-to-end-encryption-with-kafka/README.md

```diff
@@ -1,6 +1,136 @@
+
+For simplicity we're going to use single partition topics.
+
+Our goals are to make the message exchange:
+
+- secure: no one except the endpoints can decrypt the messages
+- reliable: messages are delivered eventually as long as the endpoints are alive
+
+## The example
+
+**TODO: do we want to have Rust code here, or git clone, or docker run???**
+
+
+### Git clone
+
+Ro demonstrate the example we're going to use a simple Rust program.
+
+To run it you will need a Rust compiler installed: [get it here](https://www.rust-lang.org/tools/install)
+
+Then clone the example project:
+
+```bash
+git clone ockam-network/kafka-example
+```
+
+And run the example:
+
+```bash
+cargo run --example kafka
+```
+
+Follow instuctions from the program output to start another program to establish messaging between them.
+
+
+### Rust project
+
+Ro demonstrate the example we're going to use a simple Rust program.
+
+To run it you will need a Rust compiler installed: [get it here](https://www.rust-lang.org/tools/install)
+And set up a project:
+
+```
+cargo new --lib ockam_get_started
+
+cd ockam_get_started
+
+echo 'ockam = { version = "0", features = ["ockam_transport_tcp", "ockam_vault"] }' >> Cargo.toml
+
+mkdir examples
+```
+
+Then create the example file:
+
+```bash
+touch examples/kakfa.rs
+```
+
+And put the following code there:
+
+```rust
+...
+```
+
+Then you can run the example:
+
+```bash
+cargo run --example kafka
+```
+
+Follow instuctions from the program output to start another program to establish messaging between them.
+
+### Docker image
+
+Ro demonstrate the example we're going to use a simple Rust program.
+
+We have it packaged as a docker image you can run as:
+
+```bash
+docker run ockam-network/kafka-example
+```
+
+Follow instuctions from the program output to start another program to establish messaging between them.
+
+
+## What just happened?
+
+The example program is using Ockam framework to establish connection to a cloud node Ockam Hub (**TODO: hub address**) and using an [Ockam Stream API](link to streams) creates topics in Confluent Cloud Kafka cluster.
+
+Then using these topics, the two programs establish an [Ockam Secure Channel](link to secure channels) to exchange messages.
+
+Every message you type is encrypted by Secure Channel, sent to the Hub Node and put in the Kafka topic by the Hub Node.
+
+Then it's fetched by the other program and decrypted by Secure Channel.
+
+The messages printed as `Push to the stream` and `Pull from the stream` are the messages sent to Kafka topics.
+
+Messages in those logs are the only information about the messages that a network ever sees and cannot be decrypted by anyone other then the programs.
+
+**NOTE** encryption key is transient and generated on secure channel establishment. After you exit the example programs the messages stored in Kafka will be essentially lost.
```

Comment:
s // will be essentially lost. // will be lost as they can no longer be decrypted.